logstash-core/lib/logstash/util.rb

```diff
@@ -37,18 +35,10 @@ def self.set_thread_plugin(plugin)
     Thread.current[:plugin] = plugin
   end
 
-  def self.get_thread_id(thread)
-    if RUBY_ENGINE == "jruby"
-      JRuby.reference(thread).native_thread.id
```

Comment:
the reason for moving the method to native is the `JRuby.reference` usage.
... a bit unfortunate API as it has side effects on (self) reflecting JRuby's Java classes
as of the date having this classes present some issues at runtime e.g. when looping through Class objects